build/entrypoint.d/30-reverse-proxy.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/sh
+
+# Skip creating proxy file if no proxies are configured
+if [ -z "$OSIM_NGINX_PROXY_JIRA" ]; then
+    exit 0;
+fi
+
+# Get proxy certificate
+curl --xattr $OSIM_NGINX_PROXY_CA -o /tmp/Proxy-CA.crt
+
+# Ensure no trailing slash in variable, so no duplicate trailing slash is added in proxy_pass
+OSIM_NGINX_PROXY_JIRA="${OSIM_NGINX_PROXY_JIRA%/}"
+
+echo resolver "$(awk -v ORS=' ' '$1=="nameserver" {print $2}' /etc/resolv.conf)" ";" >/etc/nginx/conf.d/resolvers.conf
+
+# Add JIRA reverse proxy endpoint
+cat <<EOF >/tmp/osim-nginx-proxy.conf
+location /proxy/jira/ {
+    # Trailing slash in proxy_pass strips the location directive prefix from the downstream URL
+    proxy_pass ${OSIM_NGINX_PROXY_JIRA}/;
+
+    proxy_ssl_trusted_certificate /tmp/Proxy-CA.crt;
+    proxy_ssl_session_reuse on; # try off if connection issues
+    proxy_ssl_server_name on; # required
+EOF
+cat <<'EOF' >>/tmp/osim-nginx-proxy.conf
+    proxy_set_header X-Real-IP $remote_addr;
+    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
+    proxy_set_header X-Forwarded-Proto https;
+}
+EOF
```

Comment:
```sh
EOF
cat <<'EOF' >>/tmp/osim-nginx-proxy.conf
```
is this intended? being that we are appending to the same file I feel like we could just use only one `cat` but I'm not 100% sure what the `'EOF'` with single quotes does
```suggestion
cat <<EOF >/tmp/osim-nginx-proxy.conf
location /proxy/jira/ {
    # Trailing slash in proxy_pass strips the location directive prefix from the downstream URL
    proxy_pass ${OSIM_NGINX_PROXY_JIRA}/;

    proxy_ssl_trusted_certificate /tmp/Proxy-CA.crt;
    proxy_ssl_session_reuse on; # try off if connection issues
    proxy_ssl_server_name on; # required
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Forwarded-Proto https;
}
EOF
```

Reply:
It's intended - for reference, the `<<EOF` construct is called a heredoc (search "Here Document" in `man bash`), and shell substitutions like `$FOO` or `${FOO}` or `$(uname)` are substituted in heredocs.
If single quotes are used around the heredoc label, like `<<'EOF'`, then the heredoc is literal and no substitutions are made.
To substitute the environment variable `${OSIM_NGINX_PROXY_JIRA}`, the first heredoc is unquoted, and to use the nginx config variables like `$remote_addr`, the second heredoc is quoted. I just had a weak preference to split the heredocs instead of escaping the nginx config variables like `\$remote_addr`

Reply:
I knew about the `<<EOF` syntax for writing files but the `<<'EOF'` is an interesting find, thanks for the explanation!